File: irrgarten_ruby/src/shield.rb
#encoding:utf-8

module Irrgarten

  require_relative 'dice'
  require_relative 'combat_element'

  class Shield < CombatElement

    def protect
      self.produce_effect
    end

    def to_s
      return "S:" + super

    end

  end #end class
end #end module

File: irrgartenjaava/src/irrgartenjaava/Shield.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgartenjaava;

/**
 *
 * @author julian
 * 
 * 
 * @brief Clase actualizada con herencia P4
 * 
 */
public class Shield extends CombatElement {

    public Shield(float protection, int uses){
        super(protection, uses);
    }
    
    public float protect(){
        return this.produceEffect();
    }
    
    @Override 
    public String toString(){
        String a_devolver="S";
        return a_devolver+=super.toString();
        
    }
    
}
